### PR TITLE
Failed Event Notifications

### PR DESCRIPTION
**Pull request by rohitp19**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [Y]
| New feature/enhancement? (use the a.x branch)      | [Y]
| Deprecations?                          | [N]
| BC breaks? (use the c.x branch)        | [N]
| Automated tests included?              | [Y] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:

This is a PR for changing the way we handle notifications for failed events within campaigns.

This PR leverages the event we already fire, `CampaignEvents::ON_EVENT_FAILED`. We're moving the notification into an event listener in which we are doing some checks to see if the given campaign event has failed for more than 5% of the contacts within the campaign. If so, notify the user. If its failed more than 10%, disable the campaign and notify the user of that.

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
1. Create a template email.
2. Create a segment and a campaign sourced from that segment.
3. Put 10 contacts in that segment (preferably faked ones that can actually receive the email)
4. Set your Email Settings to "Other SMTP" and give bogus credentials so that the email sending will fail.
5. In that campaign, create a "send email" action. Select the email you created earlier and choose it as marketing email.
6. Trigger the campaign and see that the emails failed.
7. You'll start generating a ton of entries in your notifications table that the action failed due to the contact having already received the marketing email.
8.you'll only start generating errors after 5% of the contacts have failed 
9. You'll see that only 1 notification is created until the campaign automatically unpublishes itself.
10. Ensure that you get an email stating your campaign has been unpublished. This you may not see as email settings are changed.

### CODE REVIEW

File: app/bundles/CampaignBundle/Entity/Event.php [hunk 1]
```diff
@@ -161,6 +161,11 @@ class Event implements ChannelInterface
 
     private ?\DateTimeInterface $deleted = null;
 
+    /**
+     * @var int
+     */
+    private $failedCount = 0;
```
Comment (dadarya0) [inline app/bundles/CampaignBundle/Entity/Event.php]:
```suggestion
    private int $failedCount = 0;
```

**Review by dadarya0 — CHANGES_REQUESTED** [review]
Found only 1 changes.

**Review by dadarya0 — DISMISSED** [review]

**Comment by dadarya0** [comment]
@rohitp19 failing on cs fixer.

**Review by dadarya0 — APPROVED** [review]

**Comment by Galokh66** [comment]
@rohitp19 
getting 500 server error when i click on campaign or try to edit it.
Screenshot 2024-02-22 at 12 57 48 PM
Also it breaks when i create a new campaign
Screenshot 2024-02-22 at 12 59 35 PM

**Comment by rohitp19** [comment]
@Galokh66 This PR contains a migration. Have you executed the migrations? Those can be executed using `bin/console d:m:m`

**Comment by Galokh66** [comment]
Tested and verified the notifications for email failure and campaign unpublished due to failure.
Screenshot 2024-02-22 at 3 09 16 PM
Screenshot 2024-02-22 at 3 09 25 PM

**Review by Galokh66 — APPROVED** [review]

**Comment by cstuder** [comment]
Edit: Never mind, what I'm trying to do isn't supported by the Mautic architecture. I'll use a different way to do this. Ignore my comment.

Old: 

@rohitp19 Question: Is there a way to disable the automatic disabling of campaigns on failures? I'm trying to get a plugin of mine to delay events and one possible way would be to fail and reschedule them. But the new CampaignEventSubscriber disables the campaign due to the threshold.

Details here: https://forum.mautic.org/t/how-do-i-correctly-delay-the-execution-of-a-campaign-action-in-a-plugin/34970/3

**Pull request merged**